controllers/ariticles-controller.js

```diff
@@ -17,11 +18,22 @@ exports.getArticleById = (req, res, next) => {
 exports.getAllArticles = (req, res, next) => {
   const { topic } = req.query;
 
-  selectArticles(topic)
-    .then((articles) => {
-      res.status(200).send({ articles });
-    })
-    .catch(next);
+  if (topic) {
```

Comment:
This can be handled in the model. Ideally your controller just asks for data and serves it back, goes to catch if it doesn't work. Your model should be doing the heavy lifting of checking if this exists or not.